code/modules/antagonists/nukeop/equipment/nuclearbomb.dm

```diff
@@ -501,16 +501,34 @@ This is here to make the tiles around the station mininuke change when it's arme
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 30, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 100)
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
 	var/fake = FALSE
+	var/turf/lastlocation
+	var/last_disk_move
 
 /obj/item/disk/nuclear/Initialize()
 	. = ..()
 	if(!fake)
 		GLOB.poi_list |= src
+		last_disk_move = world.time
+		START_PROCESSING(SSobj, src)
 
 /obj/item/disk/nuclear/ComponentInitialize()
 	. = ..()
 	AddComponent(/datum/component/stationloving, !fake)
 
+/obj/item/disk/nuclear/process()
+	if(fake)
+		STOP_PROCESSING(SSobj, src)
+		CRASH("A fake nuke disk tried to call process(). Who the fuck and how the fuck")
+	var/turf/newturf = get_turf(src)
+	if(istype(newturf) && lastlocation == newturf)
+		if(last_disk_move < world.time - 5000 && prob((world.time - 5000 - last_disk_move)*0.00001))
```

Comment:
since you know you only increase probability every x ticks after moving, use a stoppable timer instead of processing and update it in Moved()

Reply:
Moved() doesn't get called if the object is inside another object, which means relying on Moved() will result in odd effects, like the disk not counting as having moved if it's inside a locker being dragged around, or the disk not noticing it's moved if it's within a box inside a bag. Accommodating for these limitations of Moved() by skipping these checks if the disk isn't on a turf will mean AFK heads can accidentally cheese the entire mechanic by simply keeping the disk in their backpack while hiding in a locker somewhere.

Reply:
correct

Reply:
ugh, gayer than traps